Jenkinsfile

```diff
@@ -2,38 +2,194 @@
+                                aws ec2 describe-instance-status \
+                                --region=us-east-1 \
+                                --instance-id $INSTANCE_ID \
+                                --query 'InstanceStatuses[].InstanceStatus[].Details[].Status' \
+                                --output text
+                            """).trim()
+                        while (status != 'passed') {
+                            println "Waiting for the instance to fully Boot up...."
+                            sleep(15)
+                            status = sh(returnStdout: true, script: """
+                                aws ec2 describe-instance-status \
+                                --region=us-east-1 \
+                                --instance-id $INSTANCE_ID \
+                                --query 'InstanceStatuses[].InstanceStatus[].Details[].Status' \
+                                --output text
+                            """).trim()
+                            println "Instance Status: ${status}"
+                        }
+                    }
+                }
+                docker.image("${baseImageName}").inside {
+                    withEnv([
+                        "APP=${apk}",
+                        "platformName=android",
+                        "deviceName=127.0.0.1:5556",
+                        "MODULE=testSmoke",
+                        "TEST=SmokeTest",
+                        "appPackage=org.mozilla.fennec_",
+                        "appActivity=org.mozilla.gecko.LauncherActivity"
+                        ]) {
+                        withCredentials([file(credentialsId: 'da5f91e6-e1ca-4aac-94ea-352b6769228b', variable: 'FILE' )]) {
+                            stage('Genymotion ADB Connect') {
+                                sh'''#!/bin/bash -l
+                                    set -x
+                                    set -e
+                                    chmod 400 $FILE
+                                    ssh -v -o StrictHostKeyChecking=no -i $FILE root@$IP "setprop persist.sys.usb.config adb"
+                                    ssh -v -o StrictHostKeyChecking=no -i $FILE -NL 5556:127.0.0.1:5555 root@$IP &
+                                    $ANDROID_HOME/platform-tools/adb connect 127.0.0.1:5556
+                                    $ANDROID_HOME/platform-tools/adb wait-for-device
+                                '''
+                            }
+                        }
+                        stage('Run Tests') {
+                            timeout(60) {
+                                sh'''#!/bin/bash -l
+                                    set -x
+                                    set -e
+                                    appium &
+                                    sleep 10
+                                    export app=$PWD/mozilla-release/objdir-frontend-android/dist/$APP
```

Comment:
`withEnv`?

Reply:
The Path changes with every PR. So had to export it from the PR Directory.